From 9cbaf511c7251cc1ce84c96b5939f41d970b6e1b Mon Sep 17 00:00:00 2001
From: Zhong Hongbo <hongbo.zhong@windriver.com>
Date: Wed, 26 Feb 2014 10:18:21 +0800
Subject: [PATCH 1071/1072] kgdboc, imx: rx polling hook for the imx driver

The RX polling hook allows the debugger to hook character input so as
to allow entry to the kernel debugger with a control-c as an example.

Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 drivers/tty/serial/imx.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index d799140..4251680 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -673,6 +673,10 @@ static irqreturn_t imx_rxint(int irq, void *dev_id)
 				continue;
 		}
 
+#ifdef CONFIG_CONSOLE_POLL
+		if (sport->port.poll_rx_cb && sport->port.poll_rx_cb((unsigned char)rx))
+			return;
+#endif
 		if (uart_handle_sysrq_char(&sport->port, (unsigned char)rx))
 			continue;
 
-- 
1.7.5.4

